wadas/ui/configure_web_interface.py

```diff
@@ -0,0 +1,292 @@
+#
+# Author(s): Stefano Dell'Osa, Alessandro Palla, Cesare Di Mauro, Antonio Farina
+# Date: 2024-10-01
+# Description: Web interface users UI Module
+
+import os
+import bcrypt
+from validators import email as valid_email
+
+from PySide6.QtCore import Qt
+from PySide6.QtGui import QIcon
+from PySide6.QtWidgets import (
+    QDialog,
+    QGridLayout,
+    QLabel,
+    QLineEdit,
+    QRadioButton,
+    QScrollArea,
+    QWidget,
+    QComboBox,
+)
+
+from wadas.domain.database import DataBase, DBUser
+from wadas.ui.qt.ui_configure_web_interface import Ui_DialogConfigureWebInterface
+
+module_dir_path = os.path.dirname(os.path.abspath(__file__))
+
+class DialogConfigureWebInterface(QDialog, Ui_DialogConfigureWebInterface):
+    """Class to configure FTP server and cameras"""
+
+    def __init__(self):
+        super(DialogConfigureWebInterface, self).__init__()
+        self.ui = Ui_DialogConfigureWebInterface()
+        self.ui_user_idx = 0
+        self.removed_users = []
+        self.removed_rows = set()
+        self.roles = ["Admin", "Viewer"]
+        self.web_interface_enabled = False
+        # DB enablement status
+        self.db_enabled = bool(DataBase.get_enabled_db())
+
+        # UI
+        self.ui.setupUi(self)
+        self.setWindowIcon(QIcon(os.path.join(module_dir_path, "..", "img", "mainwindow_icon.jpg")))
+        self.ui.pushButton_remove_user.setEnabled(False)
+        self.ui.pushButton_reset_password.setEnabled(False)
+        self.ui.label_errorMessage.setStyleSheet("color: red")
+
+        # Create scrollable area for ftp camera list in FTPCamera tab
+        scroll_area = QScrollArea(self.ui.verticalLayoutWidget)
+        scroll_area.setWidgetResizable(True)
+        scroll_widget = QWidget()
+        scroll_area.setWidget(scroll_widget)
+        users_grid_layout = QGridLayout(scroll_widget)
+        users_grid_layout.setObjectName("gridLayout_users")
+        self.ui.verticalLayout_users.addWidget(scroll_area)
+        # Adding first row of camera form
+        self.add_user()
+
+        # Slots
+        self.ui.buttonBox.accepted.connect(self.accept_and_close)
+        self.ui.pushButton_add_user.clicked.connect(self.add_user)
+        self.ui.pushButton_remove_user.clicked.connect(self.remove_user)
+        self.ui.pushButton_reset_password.clicked.connect(self.reset_user_password)
+        self.ui.pushButton_stop_web_interface.clicked.connect(self.on_web_interface_stop_clicked)
+        self.ui.pushButton_start_web_interface.clicked.connect(self.on_web_interface_start_clicked)
+
+        # Init dialog
+        self.initialize_dialog()
+
+    def initialize_dialog(self):
+        """Method to initialize dialog with existing values (if any)."""
+
+        self.update_web_interface_status()
+        if self.db_enabled:
+            i = 0
+            users = DataBase.get_users()
+            for username, email, role in users:
+                if i > 0:
+                    self.add_user()
```

Comment:
for i, (username, email, role) in enumerate(DataBase.get_users()):
                if i:
                    self.add_user()